pkg/experimental/genrpm.bzl

```diff
@@ -0,0 +1,611 @@
+    non-empty, a third output will be produced, following the RPM-recommended
+    N-V-R.A format (Name-Version-Release.Architecture.rpm). Note that due to
+    the fact that rule implementations cannot access the contents of files,
+    the "version_file" and "release_file" arguments will not create an output
+    using N-V-R.A format.
+
+    This rule only functions on UNIXy platforms. The following tools must be
+    available on your system for this to function properly:
+
+    - `rpmbuild` (as specified in `rpmbuild_path`, or available in `$PATH`)
+
+    - GNU coreutils.  BSD coreutils may work, but are not tested.
+
+    """,
+    attrs = {
+        "rpm_name": attr.string(
+            doc = """Optional; RPM name override.
+
+            If not provided, the `name` attribute of this rule will be used
+            instead.
+
+            This influences values like the spec file name, and the name of the
+            output RPM.
+
+            """,
+        ),
+        "version": attr.string(
+            doc = """RPM "Version" tag.
+
+            Exactly one of `version` or `version_file` must be provided.
+            """,
+        ),
+        "version_file": attr.label(
+            doc = """File containing RPM "Version" tag.""",
+            allow_single_file = True,
+        ),
+        "release": attr.string(
+            doc = """RPM "Release" tag
+
+            Exactly one of `release` or `release_file` must be provided.
+            """,
+        ),
+        "release_file": attr.label(
+            doc = """File containing RPM "Release" tag.""",
+            allow_single_file = True,
+        ),
+        "group": attr.string(
+            doc = """Optional; RPM "Group" tag.
+
+            NOTE: some distributions (as of writing, Fedora > 17 and CentOS/RHEL
+            > 5) have deprecated this tag.  Other distributions may require it,
+            but it is harmless in any case.
+
+            """,
+        ),
+        # TODO(nacl): this should be augmented to use bazel platforms, and
+        # should not really set BuildArch.
+        "architecture": attr.string(
+            doc = """Package architecture.
+
+            This currently sets the `BuildArch` tag, which influences the output
+            architecture of the package.
+
+            Typically, `BuildArch` only needs to be set when the package is
+            known to be cross-platform (e.g. written in an interpreted
+            language), or, less common, when it is known that the application is
+            only valid for specific architectures.
+
+            When no attribute is provided, this will default to your host's
+            architecture.  This is usually what you want.
+
+            """,
+        ),
+        "license": attr.string(
+            doc = """RPM "License" tag.
+
+            The software license for the code distributed in this package.
+
+            """,
+            mandatory = True,
```

Comment:
I don't think this should be mandatory at this point. Internal things (which is the bulk of Bazel usage) usually don't think about their license terms.

Reply:
Unfortunately, `rpmbuild` bombs if it's not provided or if it's blank.

I can perhaps add a default value in there like "commercial" or "internal use only", but I feel that could be confusing.  How about I add some documentation as a compromise?

Reply:
Documentation added in 229a244